monkey/infection_monkey/propagation_credentials_repository/propagation_credentials_repository.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import time
+from itertools import chain
+from multiprocessing import get_context
+from typing import Iterable
+
+from common.credentials import Credentials
+from infection_monkey.i_control_channel import IControlChannel
+
+from .i_propagation_credentials_repository import IPropagationCredentialsRepository
+
+logger = logging.getLogger(__name__)
+
+CREDENTIALS_POLL_PERIOD_SEC = 10
+
+
+class PropagationCredentialsRepository(IPropagationCredentialsRepository):
+    """
+    Repository that stores credentials on the island and saves/gets credentials by using
+    command and control channel
+    """
+
+    def __init__(
+        self, control_channel: IControlChannel, polling_period: float = CREDENTIALS_POLL_PERIOD_SEC
```

Comment:
I'm wondering if we should use `IIslandAPIClient` here instead of `IControlChannel`. Since the introduction of `IIslandAPIClient`, I think `IControlChannel` becomes less and less valuable.

Reply:
I suppose at this point it's just serving as a facade for retrieving data from the island. I don't think I have an opinion on that at this point. I don't think it would be a difficult change to make if we wanted to get rid of `IControlChannel` entirely.

Reply:
Yeah, I feel like let's err on the side of getting rid of it. `IControlChannel` was put in place to work round the `ControlClient` in `control.py`.

Reply:
Don't forget to update the docstring.